spec/lib/scripts/collection_members_view_and_download_stats_spec.rb

```diff
@@ -0,0 +1,145 @@
+require 'rails_helper'
+require "#{Rails.root}/lib/scripts/collection_members_view_and_download_stats"
+
+RSpec.describe 'lib/scripts/collection_members_views_and_download_stats.rb', :vcr do
+  let(:user) { FactoryGirl.create(:user) }
+  let(:file) {
+    make_generic_file(user, { id: 'bogus-file-id', title: ['Test File'] })
+  }
+  let(:parent_collection) {
+    make_collection(user, { title: 'Test Collection'})
+  }
+
+  before do
+    parent_collection.members << file
+    parent_collection.save!
+  end
+
+  describe 'CollectionMembersViewAndDownloadStats' do
+    subject { CollectionMembersViewAndDownloadStats.new(parent_collection.id) }
+
+    it '#initialize' do
+      expect(subject.collection).to be_truthy
+      expect(subject.pageviews_csv_file).to be_truthy
+      expect(subject.downloads_csv_file).to be_truthy
+    end
+
+    it 'has constant STAT_TYPES' do
+      expect(subject.class::STAT_TYPES).to eq(["downloads", "pageviews"])
+    end
+
+    it 'creates the file with sanitized title for each stat type' do
+      expect(
+        File.basename(subject.pageviews_csv_file.path)
+      ).to eq("Test_Collection_pageviews_stats.csv")
+      expect(
+        File.basename(subject.downloads_csv_file.path)
+      ).to eq("Test_Collection_downloads_stats.csv")
+    end
+
+    describe '#get_stats_and_add_to_csv' do
+      context 'collection with one file' do
+        before do
+          mock_file_usage = double("File Usage")
+          allow(FileUsage).to receive(:new).with(file.id).and_return(mock_file_usage)
+          allow(mock_file_usage).to receive(:pageviews).and_return([[1570406400000, 2]])
+          allow(mock_file_usage).to receive(:downloads).and_return([[1570406400000, 1]])
```

Comment:
Nice!

Reply:
Took too long to figure out how to actually get this working, and only two minutes to make the needed changes. 🤦‍♂